backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/acct/api/AccountDimension.java

```diff
@@ -274,8 +274,11 @@ else if(value instanceof BigDecimal)
 			}
 			else
 			{
-				final String stringValue = String.valueOf(value);
-				segmentValues.put(segmentType, Strings.nullToEmpty(stringValue));
+				final String stringValue = StringUtils.asStringAndTrimBlankToNull(value);
+				if(stringValue != null)
```

Comment:
pls check if this logic change would impact `isSegmentValueSet` method.
i.e. in case the value is empty/null, before this change isSegmentValueSet would return true but now will return false.
If that's a problem, I think a quick workaround is to use `org.compiere.model.Null` as a placeholder...

Reply:
Because of `ImmutableMap.copyOf(builder.segmentValues);` it wasn't possible before already to set null values, as Immutable map doesn't allow it.
The only difference now is that UserElementStrings which before had String 'null' now don't always return true